### PR TITLE
Declare support for Python 3.13

### PR DESCRIPTION
**Pull request by edgarrmondragon**
### What does this change

The first release candidate of Python 3.13 is around the corner ([2024-07-30](https://peps.python.org/pep-0719/)), this adds it to the test matrix and declares support in the project metadata.

### What was wrong

Python 3.13 seems to be supported without any code changes, but this makes sure it's tested.

### How this fixes it

* Adds PyPI trove classifier
* Adds `3.13` to the list of versions for `actions/setup-python`

### CODE REVIEW

**Comment by edgarrmondragon** [comment]
Tests are 🟢 over on my fork: https://github.com/edgarrmondragon/faker/actions/runs/9849708765

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -154,6 +160,7 @@ jobs:
       uses: actions/setup-python@v5
       with:
         python-version: ${{ matrix.python }}
+        allow-prereleases: true
```
Comment (stefan6419846) [inline .github/workflows/ci.yml:163]: Why not using `"3.13-dev"` instead to better indicate that Python 3.13 is not yet available as a stable release?

File: tox.ini [hunk 2]
```diff
@@ -1,11 +1,11 @@
 [tox]
-envlist=py{38,39,310,311,312,py3},alpine,flake8,checkmanifest,isort,mypy,doc8
+envlist=py{38,39,310,311,312,313,py3},alpine,flake8,checkmanifest,isort,mypy,doc8
 skip_missing_interpreters = true
 
 [testenv]
 deps =
     coverage>=5.2
-    freezegun<0.4
+    freezegun<2
```
Comment (stefan6419846) [inline tox.ini:8]: How is this related to the Python 3.13 change?

Comment (edgarrmondragon) [reply tox.ini:8]: Ah, forgot to comment. freezegun 1.5.0 was made compatible with Python 3.13 via https://github.com/spulec/freezegun/commit/d75458f3f691d0c2c35559872716d0ebab738817.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (edgarrmondragon) [reply .github/workflows/ci.yml:163]: Feels easier to let it stay as `3.13` after the final release rather than coming back and changing it from `3.13-dev` to `3.13`. I don't have a strong preference, though.

File: tox.ini (hunk 2 above)
Comment (edgarrmondragon) [reply tox.ini:8]: Thoughts @stefan6419846?

Comment (stefan6419846) [reply tox.ini:8]: I am usually against needless lower pinning.

Comment (stefan6419846) [reply tox.ini:8]: Is there a specific reason for the upper limit? We tend to not have it for the other dependencies as well.

Comment (edgarrmondragon) [reply tox.ini:8]: Not really. I'd actually prefer `freezegun>=1.5.0` since that's really the hard constraint to make it work on 3.13.

Comment (edgarrmondragon) [reply tox.ini]: I've removed the upper constraint: 6c915fe0fc1505b65a247bb4c106cf9dbaf332f5

**Comment by edgarrmondragon** [comment]
Rebased and confirmed with `tox -e py313` that tests pass.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (edgarrmondragon) [inline .github/workflows/ci.yml:163]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -184,6 +197,7 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python }}
+          allow-prereleases: true
```
Comment (edgarrmondragon) [inline .github/workflows/ci.yml:200]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -226,6 +245,7 @@ jobs:
...
```
Comment (edgarrmondragon) [inline .github/workflows/ci.yml:248]:
```suggestion
```

**Comment by edgarrmondragon** [comment]
ping @fcurella 🙏

**Comment by fcurella** [comment]
Thank you!
